pkg/apis/infinispan/v1/infinispan_types.go

```diff
@@ -14,15 +14,26 @@ type InfinispanAuthInfo struct {
 
 // InfinispanSecurity info for the user application connection
 type InfinispanSecurity struct {
-	EndpointSecretName string             `json:"endpointSecretName,optional,omitempty"`
-	EndpointEncryption EndpointEncryption `json:"endpointEncryption,optional,omitempty"`
+	EndpointSecretName string              `json:"endpointSecretName,optional,omitempty"`
+	EndpointEncryption *EndpointEncryption `json:"endpointEncryption,optional,omitempty"`
 }
 
+// CertificateSourceType specifies all the possible sources for the encryption certificate
```

Comment:
// +kubebuilder:validation:Enum=Service;Secret

Reply:
Oh, looks like this will not backward compatible

Reply:
mmm yeah that's true